commit 7916134e5d9b1641effd3b6d964c806a09cfdcee
Author: Steve Dickson <steved@redhat.com>
Date:   Thu Aug 10 11:57:39 2023 -0400

    Fixed a regression in the junction code
    
    commit cdbef4e9 created a regression in the
    in the junction code by adding a O_PATH flag
    to the open() in junction_open_path()
    
    Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2213669
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/support/junction/junction.c b/support/junction/junction.c
index 0628bb0f..c1ec8ff8 100644
--- a/support/junction/junction.c
+++ b/support/junction/junction.c
@@ -63,7 +63,7 @@ junction_open_path(const char *pathname, int *fd)
 	if (pathname == NULL || fd == NULL)
 		return FEDFS_ERR_INVAL;
 
-	tmp = open(pathname, O_PATH|O_DIRECTORY);
+	tmp = open(pathname, O_DIRECTORY);
 	if (tmp == -1) {
 		switch (errno) {
 		case EPERM:
